modules/service-networking/main.tf

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_global_address" "global_addresses" {
+  for_each      = { for address in var.global_addresses : address.name => address }
+  project       = var.project_id
+  name          = each.value.name
+  purpose       = each.value.purpose
+  address_type  = each.value.type
+  prefix_length = each.value.prefix_length
+  network       = var.network.id
+}
+
+resource "google_service_networking_connection" "default" {
+  network                 = var.network.id
+  service                 = var.service
+  reserved_peering_ranges = [for name, _ in google_compute_global_address.global_addresses : name]
```

Comment:
You can write this as follows:
```
[for name in google_compute_global_address.global_addresses : name]
```

Reply:
Done!

Reply:
@q2w I apologize. Just realized it is an object. When u have one variable it will get values and u r looking for the key. 

you can rever this change.

Reply:
Done.